src/components/directory-content.jsx

```diff
@@ -412,434 +341,142 @@ const DirectoryContent = () => {
+                                          { type: event.data.type }
+                                      )
+                                  );
+                            break;
+                        case ElementType.CONTINGENCY_LIST:
+                            if (
+                                subtype ===
+                                ContingencyListType.CRITERIA_BASED.id
+                            ) {
+                                setCurrentFiltersContingencyListId(
+                                    event.data.elementUuid
+                                );
+                                setOpenDialog(subtype);
+                            } else if (
+                                subtype === ContingencyListType.SCRIPT.id
+                            ) {
+                                setCurrentScriptContingencyListId(
+                                    event.data.elementUuid
+                                );
+                                setOpenDialog(subtype);
+                            } else if (
+                                subtype ===
+                                ContingencyListType.EXPLICIT_NAMING.id
+                            ) {
+                                setCurrentExplicitNamingContingencyListId(
+                                    event.data.elementUuid
+                                );
+                                setOpenDialog(subtype);
+                            }
+                            break;
+                        case ElementType.FILTER:
+                            if (subtype === FilterType.EXPLICIT_NAMING.id) {
+                                setCurrentExplicitNamingFilterId(
+                                    event.data.elementUuid
+                                );
+                                setOpenDialog(subtype);
+                            } else if (
+                                subtype === FilterType.CRITERIA_BASED.id
+                            ) {
+                                setCurrentCriteriaBasedFilterId(
+                                    event.data.elementUuid
+                                );
+                                setOpenDialog(subtype);
+                            } else if (subtype === FilterType.EXPERT.id) {
+                                setCurrentExpertFilterId(
+                                    event.data.elementUuid
+                                );
+                                setOpenDialog(subtype);
+                            }
+                            break;
+                        default:
+                            break;
+                    }
                 }
             }
         },
         [
             childrenMetadata,
-            currentChildren,
             dispatch,
             getLink,
             handleError,
             intl,
-            selectedDirectory?.elementUuid,
+            selectedDirectory,
```

Comment:
```suggestion
            selectedDirectory?.elementUuid,
```
is it a necessary change ? we only need selectedDirectory?.elementUuid here

Reply:
I thought it was partly the cause of the abusive rerender issue I was having but reinstating it doesn't seem to break things down so fine by me